packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page/definition-tab/property-list-card/shared/property-type-form/data-type-selector/custom-expected-value-builder/shared/custom-expected-value-selector.tsx

```diff
@@ -0,0 +1,124 @@
+import { faPlus, faSearch } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon, TextField } from "@hashintel/hash-design-system";
+import {
+  Autocomplete,
+  autocompleteClasses,
+  AutocompleteProps,
+  Box,
+  inputBaseClasses,
+} from "@mui/material";
+import { useState, useRef } from "react";
+
+export type CustomExpectedValueSelectorProps<T> = {
+  inputLabel: string;
+  collapsedWidth: number;
+  options: Array<T>;
+  onChange: AutocompleteProps<T, false, false, false>["onChange"];
+  value: T[];
+} & Partial<AutocompleteProps<T, true, false, false>>;
+
+export const CustomExpectedValueSelector = <T extends any>({
+  inputLabel,
+  collapsedWidth,
+  options,
+  onChange,
+  value,
+  renderOption,
+  ...props
+}: CustomExpectedValueSelectorProps<T>) => {
+  const [autocompleteElem, setAutocompleteElem] =
+    useState<HTMLDivElement | null>(null);
+  const textFieldRef = useRef<HTMLInputElement>(null);
+
+  return (
+    <Autocomplete
+      {...props}
+      ref={(ref: HTMLDivElement) => setAutocompleteElem(ref)}
+      value={value}
+      multiple
+      popupIcon={null}
+      clearIcon={null}
+      forcePopupIcon={false}
+      selectOnFocus={false}
+      openOnFocus
+      componentsProps={{
+        popper: {
+          sx: {
+            [`.${autocompleteClasses.paper}`]: {
+              width: autocompleteElem?.getBoundingClientRect().width,
+            },
+          },
+        },
+      }}
+      clearOnBlur={false}
+      onChange={onChange}
+      options={options}
+      renderTags={() => <Box />}
+      renderOption={renderOption}
+      renderInput={({ InputProps, ...otherProps }) => {
+        const expanded =
+          textFieldRef.current === document.activeElement ||
+          textFieldRef.current?.value;
```

Comment:
I think you just moved this, but just to say this is a possible source of bugs because `renderInput` isn't necessarily called when these external values change

Reply:
It looks like `renderInput`  is being called whenever the focus and the textField value changes. maybe I'm missing something?